khal/cli.py

```diff
@@ -43,673 +54,488 @@ def setproctitle(_):
+             notstarted, format, day_format):
+    '''Print calendar with agenda.'''
+    try:
+        rows = controllers.calendar(
+            build_collection(
+                ctx.obj['conf'],
+                multi_calendar_select(ctx, include_calendar, exclude_calendar)
+            ),
+            agenda_format=format,
+            day_format=day_format,
+            once=once,
+            notstarted=notstarted,
+            daterange=daterange,
+            conf=ctx.obj['conf'],
+            firstweekday=ctx.obj['conf']['locale']['firstweekday'],
+            locale=ctx.obj['conf']['locale'],
+            weeknumber=ctx.obj['conf']['locale']['weeknumbers'],
+            monthdisplay=ctx.obj['conf']['view']['monthdisplay'],
+            hmethod=ctx.obj['conf']['highlight_days']['method'],
+            default_color=ctx.obj['conf']['highlight_days']['default_color'],
+            multiple=ctx.obj['conf']['highlight_days']['multiple'],
+            multiple_on_overflow=ctx.obj['conf']['highlight_days']['multiple_on_overflow'],
+            color=ctx.obj['conf']['highlight_days']['color'],
+            highlight_event_days=ctx.obj['conf']['default']['highlight_event_days'],
+            bold_for_light_color=ctx.obj['conf']['view']['bold_for_light_color'],
+            env={"calendars": ctx.obj['conf']['calendars']}
+        )
+        click.echo('\n'.join(rows))
+    except FatalError as error:
+        logger.debug(error, exc_info=True)
+        logger.fatal(error)
+        sys.exit(1)
 
-def _select_one_calendar_callback(ctx, option, calendar):
-    if isinstance(calendar, tuple):
-        if len(calendar) > 1:
-            raise click.UsageError(
-                'Can\'t use "--include-calendar" / "-a" more than once for this command.')
-        elif len(calendar) == 1:
-            calendar = calendar[0]
-    return _calendar_select_callback(ctx, option, calendar)
+@cli.command("list")
+@multi_calendar_option
+@click.option('--format', '-f',
+              help=('The format of the events.'))
+@click.option('--day-format', '-df',
+              help=('The format of the day line.'))
+@click.option('--once', '-o', is_flag=True,
+              help=('Print each event only once '
+                    '(even if it is repeated or spans multiple days).')
+              )
+@click.option('--notstarted', help=('Print only events that have not started.'),
+              is_flag=True)
+@click.option('--json', help=("Fields to output in json"), multiple=True)
+@click.argument('DATERANGE', nargs=-1, required=False,
+                metavar='[DATETIME [DATETIME | RANGE]]')
+@click.pass_context
+def klist(ctx, include_calendar, exclude_calendar,
+          daterange, once, notstarted, json, format, day_format):
+    """List all events between a start (default: today) and (optional)
+    end datetime."""
+    enabled_eventformatters = plugins.FORMATTERS
+    logger.warn("TODO: register user given format string as a plugin")
```

Comment:
@tcuthbert what did you plan to do here?
I'll convert this to a comment, otherwise the tests fail.